src/iosource/ManagerBase.cc

```diff
@@ -19,6 +19,10 @@
 
 #define DEFAULT_PREFIX "pcap"
 
+// This is the number of times through the loop that we will skip trying to call Poll()
+// if there is already packet data ready to process.
+static constexpr int POLL_FREQUENCY = 10;
```

Comment:
Didn't understand the comment in relation to what the code actually does with it, specifically the part about "if there is already packet data ready to process" since there's no associated check for that.  Isn't it purely "poll a packet source every 10th time" ?

Also, was thinking that rather than avoid polling every packet via an arbitrary frequency setting (like we did in the old code, too, except it was every 25th time), an alternate approach I wonder about might be: "only poll a live packet source if it failed to extract a packet the last time we processed it, otherwise assume it's ready".

Reply:
I forgot to update that comment. My original test was to see if the packet source had packet data. That won't work with live sources because we don't actually know yet, while we do for non-live sources (see the logic in `PktSrc::GetNextTimeout()`).

> only poll a live packet source if it failed to extract a packet the last time we processed it, otherwise assume it's ready

Do you mean "only poll a live packet source if we didn't process it last time through the loop"?

Reply:
> Do you mean "only poll a live packet source if we didn't process it last time through the loop"?

No, the idea was a bit different, but re-read the loop code again and see I wasn't thinking about potential to starve out other sources.  Just to re-explain the idea: I was thinking you could always process a live PktSrc until it came back empty without even bothering with a poll/timeout (i.e. allow a greedy overshoot of one iteration), but now realize we do need some "poll frequency" mechanic even with that so that other types of IOSources can still get some fair share of the loop in the case where the PktSrc is busy all the time.

So summary for this comment thread is:
* I agree with this `POLL_FREQUENCY` mechanic.
* Maybe the comment needs updating.
* A minor suggestion to improve tune-ability of `POLL_FREQUENCY` would be moving it to a script-level option (`redef` once at init-time is fine, doesn't need to be run-time dynamic) so someone can test various frequencies without having to re-compile